botocore/endpoint.py

```diff
@@ -282,6 +211,9 @@ def _needs_retry(self, attempts, operation_model, request_dict,
             time.sleep(handler_response)
             return True
 
+    def send(self, request):
```

Comment:
I'd prefer not to add a new public method to endpoint.  Why does this need to be public.  AFAICT consuming code is still using `make_request`.

Reply:
Yeah, I think private is probably better as well. I'm using `Endpoint.send` as a replacement for places we were patching out requests methods. In the long run I'd like to remove these patches and simply pass a mock/fake http_session into the endpoint though.

Reply:
Another option is to add short circuiting behavior at the endpoint layer via events, similar to how you can short circuit at the client layer via `before-call`.  We could add a `sending-http-request` event.  The main benefit there is that it should be less susceptible to churn as internals change (registering an event on a client and behavior based on returned values from events is a public API that can't change).  We could in theory make these changes to the develop branch and not have to make any (or fewer) changes to the tests in this branch.

Reply:
I don't think the event idea will work as we're not just using the mocks to fake the response, we're also using it to assert things about the request.